feat(web): add outbound WebSocket message size limit extension

gabrielbauman opened the pull request:
Add support for chunking outbound WebSocket messages when they exceed a configurable size limit. This helps avoid browser- or proxy-specific WebSocket message size restrictions while maintaining wire compatibility.

Changes:
- Add outbound_message_size_limit field to SessionBuilderInner
- Implement extension handler with safe f64->u32 casting and validation
- Update writer_task to chunk large messages when limit is set
- Add outboundMessageSizeLimit() helper function to JavaScript API


Hope this is the right-ish way to do this.

CBenoit commented on crates/ironrdp-web/src/session.rs:
```diff
@@ -212,13 +214,22 @@ impl iron_remote_desktop::SessionBuilder for SessionBuilder {
         self.clone()
     }
 
+    #[expect(clippy::cast_possible_truncation, clippy::cast_sign_loss)]
```
suggestion: There is a function `f64_to_u16_saturating_cast` at the bottom of the file, I suggest you implement something similar to isolate the lint exception further. In your case, that’s not `saturating`, so it would be `f64_to_u16_cast(value: f64) -> Option<u16>`. Alternatively, it’s possible to create a local scope.

```rust
#[expect(clippy::cast_possible_truncation, clippy::cast_sign_loss)]
let limit = { outbound_message_size_limit as u32 };
```

CBenoit reviewed (commented):
> Add support for chunking outbound WebSocket messages when they exceed a configurable size limit. This helps avoid browser- or proxy-specific WebSocket message size restrictions while maintaining wire compatibility.

Sounds useful! Out of curiosity, can you give me a few cases you discovered where this was a problem?

> Changes:
> […]
> Hope this is the right-ish way to do this.

Looks like the right way to do it to me! Thanks!
I left a few suggestions, but it’s more about code quality and style than anything.
Let me know what you think!

gabrielbauman commented:
Cleaned up based on feedback.

Replaced the if-let + continue pattern in writer_task with the suggested match statement. Makes it easier to scan since chunking is clearly the special case.

Also fixed smart quotes that had snuck in somehow, made comments proper sentences, and uncapitalized error messages per style comments.

Moved the `#[expect(clippy::cast_*)]` from function-level down to a local scope around just the f64 as u32 cast when reading the config. Keeps the suppression tight and isolated.

Added warning when someone passes a negative number or something > u32::MAX for chunk size.

Hopefully this looks ok!

CBenoit commented on crates/ironrdp-web/src/session.rs:
```diff
@@ -219,6 +221,17 @@ impl iron_remote_desktop::SessionBuilder for SessionBuilder {
+            |outbound_message_size_limit: f64| {
+                let limit = if outbound_message_size_limit >= 0.0 && outbound_message_size_limit <= f64::from(u32::MAX) {
+                    #[expect(clippy::cast_possible_truncation, clippy::cast_sign_loss)]
+                    let limit = { outbound_message_size_limit as u32 };
+                    limit
```
```suggestion
                    { outbound_message_size_limit as u32 }
```

CBenoit commented on crates/ironrdp-web/src/session.rs:
```diff
@@ -219,6 +221,16 @@ impl iron_remote_desktop::SessionBuilder for SessionBuilder {
+                    #[expect(clippy::cast_possible_truncation, clippy::cast_sign_loss)]
+                    { outbound_message_size_limit as u32 }
+                } else {
+                    warn!("Invalid outbound message size limit; fallback to unlimited");
```
```suggestion
                    warn!(outbound_message_size_limit, "Invalid outbound message size limit; fallback to unlimited");
```

CBenoit reviewed (approved):
Perfect! LGTM!
